src/include/sof/schedule/edf_schedule.h

```diff
@@ -20,6 +21,10 @@
 
 struct edf_task_pdata {
 	void *ctx;
+	uint32_t budget;	/**< cycles per systick */
```

Comment:
also explain that bugdet can be zero, and the difference in behavior between zero and non-zero values.